dowhy/causal_refuters/bootstrap_refuter.py

```diff
@@ -1,184 +1,208 @@
-                        new_data[variable].astype('category')
-
-            new_estimator = self.get_estimator_object(new_data, self._target_estimand, self._estimate)
-            new_effect = new_estimator.estimate_effect()
-            sample_estimates[index] = new_effect.value
-
-        refute = CausalRefutation(
-            self._estimate.value,
-            np.mean(sample_estimates),
-            refutation_type="Refute: Bootstrap Sample Dataset"
-        )
-
-        # We want to see if the estimate falls in the same distribution as the one generated by the refuter
-        # Ideally that should be the case as bootstrapping should not have a significant effect on the ability
-        # of the treatment to affect the outcome
-        refute.add_significance_test_results(
-            self.test_significance(self._estimate, sample_estimates)
-        )
-
-        return refute
-
-    def choose_desired_variables(self):
-        '''
-            This method provides a way to choose the confounders whose values we wish to
-            modify for finding its effect on the ability of the treatment to affect the outcome.
-        '''
-        if self._required_variables is None:
-            self._chosen_variables = None
-        elif type(self._required_variables) is int:
-            # Shuffle the confounders 
-            random.shuffle(self._target_estimand.backdoor_variables)
-            self._chosen_variables = self._target_estimand.backdoor_variables[:self._required_variables]
-        elif type(self._required_variables) is list:
-            self._chosen_variables = self._required_variables
-            
-
-        if self._chosen_variables is None:
-            self.logger.info("INFO: There are no chosen variables")
-        else:    
-            self.logger.info("INFO: The chosen variables are: " +
-                            ",".join(self._chosen_variables))
-
-    def set_diff(self, row):
-        self.categories = set(self.categories)
-        return list( self.categories - set([row]) )
-
-    def choose_random(self, row):
+from dowhy.causal_refuter import CausalRefuter, CausalRefutation
+from dowhy.causal_estimator import CausalEstimator
+import numpy as np
+import random
+from sklearn.utils import resample
+import logging
+
+class BootstrapRefuter(CausalRefuter):
+    """
+    Refute an estimate by running it on a random sample of the data containing measurement error in the 
+    confounders. This allows us to find the ability of the estimator to find the effect of the 
+    treatment on the outcome.
+    
+    It supports additional parameters that can be specified in the refute_estimate() method.
+    
+    Parameters
+    -----------
+    -'num_simulations': int, CausalRefuter.DEFAULT_NUM_SIMULATIONS by default
+    The number of simulations to be run
+    - 'sample_size': int, Size of the original data by default
+    The size of each bootstrap sample
+    - 'required_variables': int, list
```

Comment:
the docs should be updated to convey the new "-" negative selection, and that the default is True.